File: experiments/finance/gnucash/swiftui/Fragment_report_summary.swift
import SwiftUI

struct Fragment_report_summary: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      VStack(alignment: .leading, spacing: 0) {
        HStack(alignment: .center, spacing: 0) {
          Button(action: { }) {
            Text("Pie Chart")
          }
          .disabled(false)
          .frame(maxWidth: .infinity)
          Button(action: { }) {
            Text("Bar Chart")
          }
          .disabled(false)
          .frame(maxWidth: .infinity)
        }
        HStack(alignment: .center, spacing: 0) {
          Button(action: { }) {
            Text("Line Chart")
          }
          .disabled(false)
          .frame(maxWidth: .infinity)
          Button(action: { }) {
            Text("Sheet")
          }
          .disabled(false)
          .frame(maxWidth: .infinity)
        }
      }
      .frame(maxWidth: .infinity)
      .padding(.bottom, 12.0)
      Text("Expenses for last 3 months").font(.system(size: 16)).frame(maxWidth: .infinity, alignment: .leading)
      VStack(alignment: .leading, spacing: 0) {
        VStack(alignment: .leading, spacing: 0) {
        }
        .frame(maxWidth: .infinity)
        .frame(height: 250.0)
      }
      .frame(maxWidth: .infinity)
      .padding(.bottom, 12.0)
      VStack(alignment: .leading, spacing: 0) {
        VStack(alignment: .leading, spacing: 0) {
          HStack(alignment: .center, spacing: 0) {
            Text("Total Assets").font(.system(size: 18)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            Text("$ 2,500").font(.system(size: 18)).multilineTextAlignment(.trailing).frame(maxWidth: .infinity, alignment: .trailing).frame(maxWidth: .infinity)
          }
          HStack(alignment: .center, spacing: 0) {
            Text("Total Liabilities").font(.system(size: 18)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            Text("$ 2,700").font(.system(size: 18)).multilineTextAlignment(.trailing).frame(maxWidth: .infinity, alignment: .trailing).frame(maxWidth: .infinity)
          }
          HStack(alignment: .center, spacing: 0) {
            Text("Net Worth").font(.system(size: 18)).frame(maxWidth: .infinity, alignment: .leading).frame(maxWidth: .infinity)
            Text("$ -200").font(.system(size: 18)).multilineTextAlignment(.trailing).frame(maxWidth: .infinity, alignment: .trailing).frame(maxWidth: .infinity)
          }
        }
        .frame(maxWidth: .infinity)
        .padding(10.0)
      }
      .frame(maxWidth: .infinity)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
    .padding(10.0)
  }
}

struct Fragment_report_summary_Previews: PreviewProvider {
  static var previews: some View {
    Fragment_report_summary()
  }
}

File: experiments/finance/gnucash/compose/fragment_report_summary.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun fragment_report_summary() {
  Column(modifier = Modifier.fillMaxWidth().fillMaxHeight().padding(10.0.dp)) {
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(bottom = 12.0.dp)) {
      Row() {
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
          Box(modifier = Modifier.fillMaxWidth()) {
            Row(modifier = Modifier.align(Alignment.CenterStart), verticalAlignment = Alignment.CenterVertically) {
              Image(painter = painterResource(id = R.drawable.ic_data_usage_white_24dp), contentDescription = null)
              Spacer(modifier = Modifier.width(8.0.dp))
            }
            Box(modifier = Modifier.align(Alignment.Center)) {
              Text(text = "Pie Chart")
            }
          }
        }
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
          Box(modifier = Modifier.fillMaxWidth()) {
            Row(modifier = Modifier.align(Alignment.CenterStart), verticalAlignment = Alignment.CenterVertically) {
              Image(painter = painterResource(id = R.drawable.ic_equalizer_white_24dp), contentDescription = null)
              Spacer(modifier = Modifier.width(8.0.dp))
            }
            Box(modifier = Modifier.align(Alignment.Center)) {
              Text(text = "Bar Chart")
            }
          }
        }
      }
      Row() {
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
          Box(modifier = Modifier.fillMaxWidth()) {
            Row(modifier = Modifier.align(Alignment.CenterStart), verticalAlignment = Alignment.CenterVertically) {
              Image(painter = painterResource(id = R.drawable.ic_trending_up_white_24dp), contentDescription = null)
              Spacer(modifier = Modifier.width(8.0.dp))
            }
            Box(modifier = Modifier.align(Alignment.Center)) {
              Text(text = "Line Chart")
            }
          }
        }
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
          Box(modifier = Modifier.fillMaxWidth()) {
            Row(modifier = Modifier.align(Alignment.CenterStart), verticalAlignment = Alignment.CenterVertically) {
              Image(painter = painterResource(id = R.drawable.ic_subject_white_24dp), contentDescription = null)
              Spacer(modifier = Modifier.width(8.0.dp))
            }
            Box(modifier = Modifier.align(Alignment.Center)) {
              Text(text = "Sheet")
            }
          }
        }
      }
    }
    Text(text = "Expenses for last 3 months", modifier = Modifier.fillMaxWidth().wrapContentHeight(), fontSize = 16.sp)
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(bottom = 12.0.dp)) {
      Column(modifier = Modifier.fillMaxWidth().height(250.0.dp)) {
      }
    }
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
      Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(10.0.dp)) {
        Row() {
          Text(text = "Total Assets", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 18.sp)
          Text(text = "$ 2,500", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 18.sp, textAlign = TextAlign.End)
        }
        Row() {
          Text(text = "Total Liabilities", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 18.sp)
          Text(text = "$ 2,700", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 18.sp, textAlign = TextAlign.End)
        }
        Row() {
          Text(text = "Net Worth", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 18.sp)
          Text(text = "$ -200", modifier = Modifier.wrapContentHeight().weight(1.0f), fontSize = 18.sp, textAlign = TextAlign.End)
        }
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun fragment_report_summary_preview() {
    MyApplicationTheme {
        fragment_report_summary()
    }
}
